src/embedded/embeddedSessionManager.ts

```diff
@@ -75,35 +77,38 @@ export class EmbeddedSessionManager {
     }
 
     if (this.impressions.size) {
+      const sessionPayload: EmbeddedSessionRequestPayload = {
+        session: {
+          start: this.session.start?.getTime(),
+          end: new Date().getTime(),
+          id: this.session.id
+        },
+        impressions: this.session.impressions,
```

Comment:
@mprew97 I think here we use this.impressions as we set the data in instance variable instead of session one in line: 104

Reply:
@hardikmashru do you mean `this.getImpressionList` should be used instead? Not sure if i follow you.

Reply:
gotcha